src/commcare_cloud/commands/terraform/aws.py

```diff
@@ -150,3 +159,116 @@ def run(self, args, unknown_args):
 
         with open(environment.paths.inventory_ini, 'w') as f:
             f.write(out_string)
+
+
+DEFAULT_SIGN_IN_DURATION_MINUTES = 30
+
+
+class AwsSignIn(CommandBase):
+    command = 'aws-sign-in'
+    help = """
+        Use your MFA device to "sign in" to AWS for <duration> minutes (default {})
+
+        This will store the temporary session credentials in ~/.aws/credentials
+        under a profile named with the pattern "<aws_profile>:profile".
+        After this you can use other AWS-related commands for up to <duration> minutes
+        before having to sign in again.
+    """.format(DEFAULT_SIGN_IN_DURATION_MINUTES)
+
+    arguments = [
+        Argument('--duration-minutes', type=int, default=DEFAULT_SIGN_IN_DURATION_MINUTES, help="""
+            Stay signed in for this many minutes
+        """)
+    ]
+
+    def run(self, args, unknown_args):
+        environment = get_environment(args.env_name)
+        duration_minutes = args.duration_minutes
+        aws_profile = environment.terraform_config.aws_profile
+        aws_sign_in(aws_profile, duration_minutes, force_new=True)
+
+
+@memoized
+def aws_sign_in(aws_profile, duration_minutes=DEFAULT_SIGN_IN_DURATION_MINUTES,
+                force_new=False):
+    """
+    Create a temp session through MFA for a given aws profile
+
+    :param aws_profile: The name of an existing aws profile to create a temp session for
+    :param duration_minutes: How long to set the session expiration if a new one is created
+    :param force_new: If set to True, creates new credentials even if valid ones are found
+    :return: The name of temp session profile.
+             (Always the passed in profile followed by ':session')
+    """
+    aws_session_profile = '{}:session'.format(aws_profile)
+    if not force_new \
+            and _has_valid_session_credentials(aws_session_profile):
+        return aws_session_profile
+
+    default_username = getpass.getuser()
+    username = input("Enter username associated with credentials [{}]: ".format(
+        default_username)) or default_username
```

Comment:
On one machine I use regularly, the value returned by `getpass.getuser()` is not the same as my username. Would be nice to be able to set this in an environment variable or something to skip this prompt.

Reply:
Interesting. That would probably be useful in a couple other places, one at least that I can think of. Can I make an issue for that and fix shortly in a followup?